packages/core/src/application.ts

```diff
@@ -61,9 +62,95 @@ export class Application extends Context {
    */
   controller(controllerCtor: ControllerClass, name?: string): Binding {
     name = name || controllerCtor.name;
-    return this.bind(`controllers.${name}`)
+    return this.bind(`${CoreBindings.CONTROLLERS}.${name}`)
       .toClass(controllerCtor)
-      .tag('controller');
+      .tag(CoreBindings.CONTROLLERS_TAG);
+  }
+
+  /**
+   * Register a booter class with this application.
+   *
+   * @param booterCls {Function} The booter class (constructor function).
+   * @param {string=} name Optional booter name, defaults to the class name.
+   * @return {Binding} The newly created binding, you can use the reference to
+   * further modify the binding, e.g. lock the value to prevent further
+   * modifications.
+   *
+   * ```ts
+   * class MyBooter implements Booter {}
+   * app.booter(MyBooter);
+   * ```
+   */
+  booter<T extends Booter>(booterCls: Constructor<T>, name?: string): Binding {
+    name = name || booterCls.name;
+    return this.bind(`${CoreBindings.BOOTERS}.${name}`)
+      .toClass(booterCls)
+      .tag(CoreBindings.BOOTERS_TAG)
+      .inScope(BindingScope.SINGLETON);
+  }
+
+  /**
+   * Register an array of booter classes with this application.
+   * Each Booter added in this way will automatically be named based on the
+   * class constructor name with the "booters." prefix.
+   *
+   * If you wish to control the binding keys for particular booter instances,
+   * use the app.booter function instead.
+   *
+   * @param {Constructor<Booter>[]} booterArr {Function} An array of Booter
+   * constructors.
+   * @return {Binding[]} An array of bindings for the registered Booter classes.
+   *
+   * ```ts
+   * app.booters([ControllerBooter, RepositoryBooter]);
+   * ```
+   */
+  booters<T extends Booter>(booterArr: Constructor<T>[]): Binding[] {
+    return booterArr.map(booterCls => this.booter(booterCls));
+  }
+
+  /**
+   * Function is responsible for calling all registered Booter classes that
+   * are bound to the Application instance. Each phase of an instance must
+   * complete before the next phase is started.
+   * @param {BootOptions} bootOptions Options for boot. Bound for Booters to
+   * receive via Dependency Injection.
+   */
+  async boot(bootOptions?: BootOptions) {
```

Comment:
Majority of this logic should be refactored into a `BootStrapper` class in `boot` package as the extension point for all booters. `app.boot` is just a sugar method.

Reply:
Also as I mentioned earlier, the boot configuration is an internal business of the `Application` class that should not be leaking to Application's public API. The `boot` method should not be accepting any assembly-related config.